src/server/pkg/ppsutil/util.go

```diff
@@ -259,20 +259,41 @@ func PipelineReqFromInfo(pipelineInfo *ppsclient.PipelineInfo) *ppsclient.Create
 	}
 }
 
+type pipelineDecoder interface {
+	Decode(v interface{}) error
+}
+
+// NoParseErr is returned when no parser in PipelineManifestReader is able to
+// parse a pipeline spec.
+var NoParseErr = errors.New("no decoders are able to parse pipeline spec")
+
 // PipelineManifestReader helps with unmarshalling pipeline configs from JSON. It's used by
 // 'create pipeline' and 'update pipeline'
+//
+// Note that Go's json decoder is able to parse text that gopkg.in/yaml.v3
+// cannot (multiple json documents) so we currently attempt to parse a pipeline
+// with both parsers concurrently and keep whichever parse succeeds. Hopefully
+// multi-document support is added to our yaml parser and we can remove the json
+// parser.
 type PipelineManifestReader struct {
-	buf     bytes.Buffer
-	decoder *json.Decoder
+	ds []pipelineDecoder
```

Comment:
I feel like `ds` should be something more descriptive, like `decoders`

Reply:
Done